### PR TITLE
From Lee Matos: Customize library icons

### PR DESCRIPTION
**Pull request by Pegasus-RPG**
I'm just proposing this from Bug 1327614 to get it on our radar. This will likely need updates before it can be merged. It's intended to allow customization of the icons in the library side bar, both custom icons as well as changing the color of the stock ones.

### CODE REVIEW

**Comment by rryan** [comment]
We should consider using Qt search paths to define an "icons" namespace or something and allow the skin to add a search path that overrides it.
http://doc.qt.io/qt-4.8/qdir.html#setSearchPaths

File: src/library/analysisfeature.cpp [hunk 1]
```diff
@@ -59,8 +60,21 @@ QVariant AnalysisFeature::title() {
     return m_Title;
 }
 
+QString AnalysisFeature::getIconName(){
+    return "ic_library_prepare.png";
+}
+
 QIcon AnalysisFeature::getIcon() {
```
Comment (daschuer) [inline src/library/analysisfeature.cpp]: This should go to the base class.

File: src/library/analysisfeature.cpp [hunk 2]
```diff
@@ -59,8 +60,21 @@ QVariant AnalysisFeature::title() {
...
-    return QIcon(":/images/library/ic_library_prepare.png");
+    QString current_icon = getIconName();
+    SkinLoader sl(m_pConfig);
+    QString current_skin = sl.getConfiguredSkinPath();
+    current_skin.append("/ic_library_prepare.png");
```
Comment (daschuer) [inline src/library/analysisfeature.cpp]: Why not 
current_skin.append("/" + current_icon);

**Comment by rryan** [comment]
Howdy @leematos -- are you still interested in working on this?

**Comment by daschuer** [comment]
The changeable color is part of https://github.com/mixxxdj/mixxx/pull/1117

**Comment by leematos** [comment]
@rryan whoa! blast from the past. I'd love to, but I don't see myself having time in the next few months to make any meaningful movement. <3

**Comment by Be-ing** [comment]
This pull request has not been updated in years so I am closing it to reduce the clutter of open PRs. It is also not clear that this is relevant anymore.

**Pull request closed without merging**